mmengine/visualization/vis_backend.py

```diff
@@ -816,3 +816,142 @@ def _flatten(self, d, parent_key='', sep='.') -> dict:
+
+    @property  # type: ignore
+    @force_init_env
+    def experiment(self):
+        """Return clearml object."""
+        return self._clearml
+
+    @force_init_env
+    def add_config(self, config: Config, **kwargs) -> None:
+        """Record the config to clearml.
+
+        Args:
+            config (Config): The Config object
+        """
+        self._cfg = config
+        self._task.connect_configuration(vars(config))
+
+    @force_init_env
+    def add_image(self,
+                  name: str,
+                  image: np.ndarray,
+                  step: int = 0,
+                  **kwargs) -> None:
+        """Record the image to clearml.
+
+        Args:
+            name (str): The image identifier.
+            image (np.ndarray): The image to be saved. The format
+                should be RGB.
+            step (int): Global step value to record. Default to 0.
+        """
+        self._logger.report_image(
+            title=name, series=name, iteration=step, image=image)
+
+    @force_init_env
+    def add_scalar(self,
+                   name: str,
+                   value: Union[int, float, torch.Tensor, np.ndarray],
+                   step: int = 0,
+                   **kwargs) -> None:
+        """Record the scalar data to clearml.
+
+        Args:
+            name (str): The scalar identifier.
+            value (int, float, torch.Tensor, np.ndarray): Value to save.
+            step (int): Global step value to record. Default to 0.
+        """
+        self._logger.report_scalar(
+            title=name, series=name, value=value, iteration=step)
+
+    @force_init_env
+    def add_scalars(self,
+                    scalar_dict: dict,
+                    step: int = 0,
+                    file_path: Optional[str] = None,
+                    **kwargs) -> None:
+        """Record the scalar's data to clearml.
+
+        Args:
+            scalar_dict (dict): Key-value pair storing the tag and
+                corresponding values.
+            step (int): Global step value to record. Default to 0.
+            file_path (str, optional): Useless parameter. Just for
+                interface unification. Default to None.
+        """
+        assert 'step' not in scalar_dict, 'Please set it directly ' \
+                                          'through the step parameter'
+        for key, value in scalar_dict.items():
+            self._logger.report_scalar(
+                title=key, series=key, value=value, iteration=step)
+
+    def close(self) -> None:
+        """Close the clearml."""
+        file_paths: List[str] = list()
+        for filename in scandir(self._cfg.work_dir, self._artifact_suffix,
+                                False):
+            file_path = osp.join(self._cfg.work_dir, filename)
+            file_paths.append(file_path)
+
+        for file_path in file_paths:
+            self._task.upload_artifact(os.path.basename(file_path), file_path)
+
+        if hasattr(self, '_clearml'):
```

Comment:
If `_clearml` is not set, `_task` neither

Reply:
Error catch added

Reply:
![image](https://github.com/open-mmlab/mmengine/assets/57566630/b4e68683-c8c3-4ffd-b25c-4a1d00ae1750)
Hi, I see that `_save_dir` is not used here, are there other concerns?

Reply:
https://github.com/open-mmlab/mmengine/pull/1091#discussion_r1191933389

Reply:
Got it, but I think we should skip uploading the files when `self._cfg` is not defined, otherwise `close` method will raise an error if `add_config` is not called.

Reply:
I think a catch like this is good enough. 
 https://github.com/gachiemchiep/mmengine/blob/fa6fa5939935f4c4e0b1e07c1f2e788b00086a28/mmengine/visualization/vis_backend.py#L947